pyg_lib/test/cpp/test_cuda_version.cpp

```diff
@@ -0,0 +1,8 @@
+#include <gtest/gtest.h>
+
+#include <pyg/library.h>
+
+TEST(CudaVersionTest, BasicAssertions) {
+  // CPU detect no cuda
+  EXPECT_EQ(pyg::cuda_version(), (int64_t)-1);
```

Comment:
I assume this test will fail on a gpu machine?

Reply:
I will add another test flag to specify if cuda is enabled for the test